mamba/mamba/api.py

```diff
@@ -0,0 +1,167 @@
+        repo.set_installed()
+
+    def replace_channels(self):
+        self.local_index = get_index(
+            (self.output_folder,), platform=self.platform, prepend=False
+        )
+
+        for _, v in self.local_repos.items():
+            v.clear(True)
+
+        start_prio = len(self.channels) + len(self.index)
+        for subdir, channel in self.local_index:
+            if not subdir.loaded():
+                continue
+
+            # support new mamba
+            if isinstance(channel, dict):
+                channelstr = channel["url"]
+                channelurl = channel["url"]
+            else:
+                channelstr = str(channel)
+                channelurl = channel.url(with_credentials=True)
+
+            cp = subdir.cache_path()
+            if cp.endswith(".solv"):
+                os.remove(subdir.cache_path())
+                cp = cp.replace(".solv", ".json")
+
+            self.local_repos[channelstr] = libmambapy.Repo(
+                self.pool, channelstr, cp, channelurl
+            )
+
+            self.local_repos[channelstr].set_priority(start_prio, 0)
+            start_prio -= 1
+
+    def solve(self, specs, pkg_cache_path=None):
+        """Solve given a set of specs.
+        Parameters
+        ----------
+        specs : list of str
+            A list of package specs. You can use `conda.models.match_spec.MatchSpec`
+            to get them to the right form by calling
+            `MatchSpec(mypec).conda_build_form()`
+        Returns
+        -------
+        transaction : libmambapy.Transaction
+            The mamba transaction.
+        Raises
+        ------
+        RuntimeError :
+            If the solver did not find a solution.
+        """
+        solver_options = [(libmambapy.SOLVER_FLAG_ALLOW_DOWNGRADE, 1)]
+        api_solver = libmambapy.Solver(self.pool, solver_options)
+        _specs = specs
+
+        api_solver.add_jobs(_specs, libmambapy.SOLVER_INSTALL)
+        success = api_solver.solve()
+
+        if not success:
+            error_string = "Mamba failed to solve:\n"
+            for s in _specs:
+                error_string += f" - {s}\n"
+            error_string += "\nwith channels:\n"
+            for c in self.channels:
+                error_string += f" - {c}\n"
+            pstring = api_solver.problems_to_str()
+
+            pstring = "\n".join(["- " + el for el in pstring.split("\n")])
+            error_string += f"\nThe reported errors are:\n{pstring}"
+            print(error_string)
+            raise RuntimeError("Solver could not find solution." + error_string)
+
+        if pkg_cache_path is None:
+            # use values from conda
+            pkg_cache_path = self.context.pkgs_dirs
+
+        package_cache = libmambapy.MultiPackageCache(pkg_cache_path)
+        return libmambapy.Transaction(api_solver, package_cache)
+
+
+def install(env_name: str, specs: list = [], channels: list = [], target_platform: str = None):
```

Comment:
Not sure what to do if `target_platform` is not provided? Is there a way to know the target platform for which the env was previously created? Also I guess a meaningful default target platform for creating envs should be the current platform?